### DISCUSSION TITLE
Phantom processing queue

### DISCUSSION THREAD

**Discussion by lloydbayley** [mcguirepr89/BirdNET-Pi#823] [Q&A] [answered]
Well, it came to pass that the USB flash drive I was using instead of an SD card because it would be more reliable, was a BAD idea and completely untrue to boot.

I setup a new instance, restored my database and the vast hoard of recordings that I insist on keeping because....reasons... :)

Now, all seems ok but the main screen has:

Your system is currently processing a backlog of audio. This can take several hours before normal functionality of your BirdNET-Pi resumes.

Now this would be fine if it had, however, it's had no new detections yet as it's night time here.

My impression is that I have a leftover file from when it crashed (yes, it was processing a backlog) that I've somehow restored and it's acting on the existance of that file.

I've delved inside overview.php on line 228 and manually ran the queries on the DB to no matches.

Is there a rogue file I've missed somewhere?

**Comment by ehpersonal38** [comment]
Hey Lloyd, sorry that message made you think something was wrong with your system. Actually I don't think there is. That message will appear when the code that shows the "most recent detection" can't find a detection within the last 15 database entries that has a valid spectrogram image. 

So what must've happened is you restored your recordings and database but not the associated spectrogram images?

I suppose I should clarify the message a bit more! Do you have any suggestions on what I can change it to?

**Reply by lloydbayley** [reply]
Hmm...interesting.
I copied over the whole BirdSong dir, so that should include everything shouldn't it?
Where should I be looking?

**Reply by ehpersonal38** [reply]
Yeah it should... make sure your png files are intact in any directory in `~/BirdSongs/Extracted/By_Date/{date}/{species}`
![filezilla_ItPInCoGPi](https://user-images.githubusercontent.com/103586016/229843543-2dd87a92-98f0-4c8f-9e80-e83dcb1830dd.png)

**Comment by heyho74** [comment]
I had this message for several hours with a brand new install at the weekend.  No idea how long it took to resume normality, but the following day it had sorted itself out.

**Comment by blogdemoi** [comment]
I also have this message sometimes. Just do nothing and it will be back to normal soon :-)

**Comment by lloydbayley** [comment]
Yes, it corrected itself overnight when the first birds were recorded this morning. I have a feeling my restore substituted some characters in the filenames from the backup. Will investigate because it will preclude me from listening to old recordings.

**Comment by lloydbayley** [comment, marked as answer]
Yeh, I was right. Somehow, it seems that now I have filenames like this:

<img width="730" alt="image" src="https://user-images.githubusercontent.com/21309308/230122660-9f870e62-03a5-4351-b95d-664dc776ac26.png">

Which is just bloody wonderful. WHY it chose to replace the colons in the time with whatever-the-seven-kinds-of-blancmange THOSE are, is beyond me.

Now I have to create a little script to find and fix it. BAH.
I do this stuff all the time for managing data, so it won't be too tedious I hope. Regex is your friend...well, mostly.

Just thought I'd let everyone know what it was in the end.

**Reply by lloydbayley** [reply]
For anyone interested in quick snippets:

```
find . -depth -name '**' -execdir bash -c 'mv -- "$1" "${1///:}"' bash {} \;
```
did the trick!

<img width="398" alt="image" src="https://user-images.githubusercontent.com/21309308/230129702-7157cf3c-bc82-4dc5-b1ab-6ad5f172e3ee.png">
